ddtrace/span.py

```diff
@@ -214,19 +214,17 @@ def to_dict(self):
 
         return d
 
-    def set_traceback(self):
-        """ If the current stack has a traceback, tag the span with the
-            relevant error info.
-
-            >>> span.set_traceback()
-
-            is equivalent to:
-
-            >>> exc = sys.exc_info()
-            >>> span.set_exc_info(*exc)
+    def set_traceback(self, limit=20):
+        """ If the current stack has an exception, tag the span with the
+            relevant error info. If not, set the span to the current python stack.
         """
         (exc_type, exc_val, exc_tb) = sys.exc_info()
-        self.set_exc_info(exc_type, exc_val, exc_tb)
+
+        if (exc_type and exc_val and exc_tb):
+            self.set_exc_info(exc_type, exc_val, exc_tb)
+        else:
+            tb = ''.join(traceback.format_stack(limit=limit))
+            self.set_tag(errors.ERROR_STACK, tb)  # FIXME[gabin] Change "error.stack" into "python.stack"
```

Comment:
oh if it's something we should schedule for later it's ok. Better keeping the `FIXME` than sending twice the error stack so we're good here 👍